apps/user.service/src/schemas/user.schema.ts

```diff
@@ -0,0 +1,26 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { HydratedDocument } from 'mongoose';
+
+export type UserDocument = HydratedDocument<User>;
+@Schema({ timestamps: true })
+export class User {
+  @Prop({ type: String, required: true, minlength: 2, maxlength: 20 })
```

Comment:
Nice! This is describing the user :) how about runtime validation for free ;) 
Search google for "class validators" and add it to the project.